tests/integration/lba/runner.py

```diff
@@ -0,0 +1,131 @@
+
+    args.extend(queries)
+    args.append(str(expected))
+
+    subprocess.run(args).check_returncode()
+
+
+def execute_test(memgraph_binary: str, tester_binary: str, filtering_binary: str) -> None:
+    storage_directory = tempfile.TemporaryDirectory()
+    memgraph_args = [memgraph_binary, "--data-directory", storage_directory.name]
+
+    def execute_admin_queries(queries):
+        return execute_tester(
+            tester_binary, queries, should_fail=False, check_failure=True, username="admin", password="admin"
+        )
+
+    def execute_user_queries(queries, should_fail=False, failure_message="", check_failure=True):
+        return execute_tester(tester_binary, queries, should_fail, failure_message, "user", "user", check_failure)
+
+    # Start the memgraph binary
+    memgraph = subprocess.Popen(list(map(str, memgraph_args)))
+    time.sleep(0.1)
+    assert memgraph.poll() is None, "Memgraph process died prematurely!"
+    wait_for_server(7687)
+
+    # Register cleanup function
+    @atexit.register
+    def cleanup():
+        if memgraph.poll() is None:
+            memgraph.terminate()
+        assert memgraph.wait() == 0, "Memgraph process didn't exit cleanly!"
+
+    # Prepare all users
+    execute_admin_queries(
+        [
+            "CREATE USER admin IDENTIFIED BY 'admin'",
+            "GRANT ALL PRIVILEGES TO admin",
+            "CREATE USER user IDENTIFIED BY 'user'",
+            "GRANT LABELS :label1, :label2, :label3 TO user",
+            "GRANT EDGE_TYPES :edgeType1, :edgeType2 TO user",
+            "MERGE (l1:label1 {name: 'test1'})",
+            "MERGE (l2:label2  {name: 'test2'})",
+            "MATCH (l1:label1),(l2:label2) WHERE l1.name = 'test1' AND l2.name = 'test2' CREATE (l1)-[r:edgeType1]->(l2)",
+            "MERGE (l3:label3  {name: 'test3'})",
+            "MATCH (l1:label1),(l3:label3) WHERE l1.name = 'test1' AND l3.name = 'test3' CREATE (l1)-[r:edgeType2]->(l3)",
+            "MERGE (mix:label3:label1  {name: 'test4'})",
+            "MATCH (l1:label1),(mix:label3) WHERE l1.name = 'test1' AND mix.name = 'test4' CREATE (l1)-[r:edgeType2]->(mix)",
+        ]
+    )
+
+    # Run the test with all combinations of permissions
+    print("\033[1;36m~~ Starting edge filtering test ~~\033[0m")
+    execute_filtering(filtering_binary, ["MATCH (n)-[r]->(m) RETURN n,r,m"], 3, "user", "user")
+    execute_admin_queries(["DENY EDGE_TYPES :edgeType1 TO user"])
+    execute_filtering(filtering_binary, ["MATCH (n)-[r]->(m) RETURN n,r,m"], 2, "user", "user")
+    execute_admin_queries(["GRANT EDGE_TYPES :edgeType1 TO user", "DENY LABELS :label3 TO user"])
+    execute_filtering(filtering_binary, ["MATCH (n)-[r]->(m) RETURN n,r,m"], 1, "user", "user")
+    execute_admin_queries(["REVOKE LABELS * FROM user", "REVOKE EDGE_TYPES * FROM user"])
+    execute_filtering(filtering_binary, ["MATCH (n)-[r]->(m) RETURN n,r,m"], 0, "user", "user")
+    print("\033[1;36m~~ Finished edge filtering test ~~\033[0m\n")
```

Comment:
Add this one as well (after having all privileges), just thought of one more, nothing that breaks:
    execute_admin_queries(["DENY LABELS :label1 TO user"])
    execute_filtering(filtering_binary, ["MATCH (n)-[r]->(m) RETURN n,r,m"], 0, "user", "user")